src/testdrive/action/sql.rs

```diff
@@ -77,34 +80,26 @@ impl Action for SqlAction {
     fn redo(&self, state: &mut State) -> Result<(), String> {
         let query = &self.cmd.query;
         print_query(&query);
-        let max = match self.stmt {
-            // TODO(benesch): this is horrible. SELECT needs to learn to wait
-            // until it's up to date.
-            Statement::Query { .. } => env::var("MZ_TD_MAX_RETRIES")
-                .unwrap_or_else(|_| "7".into())
-                .parse()
-                .map_err(|e| format!("invalid MZ_TD_MAX_RETRIES: {}", e))?,
-            _ => 0,
-        };
-        let mut i = 0;
+        let mut total_backoff = 0;
+        let mut backoff = std::cmp::min(100, self.timeout);
         loop {
-            let backoff = Duration::from_millis(100 * 2_u64.pow(i));
             match self.try_redo(&mut state.pgclient, &query) {
                 Ok(()) => {
-                    if i > 0 {
+                    if total_backoff > 0 {
                         println!();
                     }
                     println!("rows match; continuing");
                     break;
                 }
                 Err(err) => {
-                    if i == 0 {
+                    if total_backoff == 0 {
                         print!(
                             "rows didn't match; sleeping to see if dataflow catches up {:?}",
                             backoff
                         );
                         io::stdout().flush().unwrap();
-                    } else if i < max {
+                    } else if total_backoff < self.timeout {
+                        backoff = std::cmp::min(backoff * 2, self.timeout - total_backoff);
```

Comment:
Per the [Rust style guide](https://doc.rust-lang.org/1.0.0/style/style/imports.html#prefer-fully-importing-types/traits-while-module-qualifying-functions.), please `use std::cmp;` at the beginning of the file and use `cmp::min` here.